collect_app/src/main/java/org/odk/collect/android/formhierarchy/FormHierarchyFragment.java

```diff
@@ -603,21 +175,25 @@ public void refreshView() {
      * @see #refreshView()
      */
     private void refreshView(boolean isGoingUp) {
+        FormHierarchyLayoutBinding binding = FormHierarchyLayoutBinding.bind(requireView());
+        ImageView groupIcon = binding.groupIcon;
+        TextView groupPathTextView = binding.pathtext;
+        RecyclerView recyclerView = binding.list;
+
         try {
             FormController formController = formEntryViewModel.getFormController();
 
             // Save the current index so we can return to the problematic question
             // in the event of an error.
-            currentIndex = formController.getFormIndex();
+            formHierarchyViewModel.setCurrentIndex(formController.getFormIndex());
 
-            elementsToDisplay = new ArrayList<>();
+            ArrayList<HierarchyItem> elementsToDisplay = new ArrayList<>();
```

Comment:
You should use the `List` interface here:
`List<HierarchyItem> elementsToDisplay = new ArrayList<>();
`